File: 1012 - area/area.cpp
#include <iostream>
#include <iomanip>

using namespace std;

class Geometrico{

    private:
        double areadotrianguloretangulo( double baseA, double alturaC  ){
            return ((baseA * alturaC) / 2.0 );
        }

        double areadocirculo( double raioC ){
            return ( 3.14159 * raioC * raioC);
        }

        double areadotrapezio( double baseA, double baseB, double alturaC )
        {
            return (((baseA + baseB) * alturaC ) / 2.0 );
        }

        double areadoquadrado( double ladoB )
        {
            return (ladoB * ladoB);
        }

        double areadoretangulo( double ladoA, double ladoB )
        {
            return (ladoA * ladoB);
        }
    
    public: 
        void show(){
            double a, b, c;
            cin >> a >> b >> c;
            cout << "TRIANGULO: " << fixed << setprecision(3) << areadotrianguloretangulo(a, c) << "\n";
            cout << "CIRCULO: " << fixed << setprecision(3) <<  areadocirculo(c) << "\n";
            cout << "TRAPEZIO: " << fixed << setprecision(3) <<  areadotrapezio(a, b, c) << "\n";
            cout << "QUADRADO: " << fixed << setprecision(3) <<  areadoquadrado(b) << "\n";
            cout << "RETANGULO: " << fixed << setprecision(3) <<  areadoretangulo(a, b) << "\n";
        }


};

int main(void)
{
    Geometrico foo;
    foo.show();
    return 0;
}

File: 1012 - area/area.hs
import Text.Printf
import System.IO
import Prelude

areadotrianguloretangulo :: Double -> Double -> Double
areadotrianguloretangulo baseA alturaC = (baseA * alturaC) / 2.0

areadocirculo:: Double -> Double
areadocirculo raioC = 3.14159 * raioC * raioC

areadotrapezio :: Double -> Double -> Double -> Double
areadotrapezio baseA baseB alturaC = ((baseA + baseB) * alturaC ) / 2.0 

areadoquadrado :: Double -> Double
areadoquadrado ladoB = ladoB * ladoB

areadoretangulo :: Double -> Double -> Double
areadoretangulo ladoA ladoB = ladoA * ladoB

main = do
    linha <- getLine
    let foo = words linha
        a = read (foo !! 0) :: Double
        b = read (foo !! 1) :: Double
        c = read (foo !! 2) :: Double

    let varareadotrianguloretangulo = areadotrianguloretangulo a c
    let varareadocirculo = areadocirculo c
    let varareadotrapezio = areadotrapezio a b c
    let varareadoquadrado = areadoquadrado b
    let varareadoretangulo = areadoretangulo a b

    printf "TRIANGULO: %.3f\n" varareadotrianguloretangulo
    printf "CIRCULO: %.3f\n" varareadocirculo
    printf "TRAPEZIO: %.3f\n" varareadotrapezio
    printf "QUADRADO: %.3f\n" varareadoquadrado
    printf "RETANGULO: %.3f\n" varareadoretangulo       